src/connectors/hits-per-page/connectHitsPerPage.ts

```diff
@@ -92,39 +90,28 @@ export type HitsPerPageRendererOptions<
    * Indicates whether or not the search has results.
    */
   hasNoResults: boolean;
-} & RendererOptions<THitsPerPageWidgetParams>;
-
-export type HitsPerPageRenderer<
-  THitsPerPageWidgetParams extends HitsPerPageConnectorParams = HitsPerPageConnectorParams
-> = Renderer<HitsPerPageRendererOptions<THitsPerPageWidgetParams>>;
-
-export type HitsPerPageWidgetFactory<
-  THitsPerPageWidgetParams extends HitsPerPageConnectorParams = HitsPerPageConnectorParams
-> = WidgetFactory<HitsPerPageConnectorParams & THitsPerPageWidgetParams>;
-
-type HitsPerPageConnector<
-  THitsPerPageWidgetParams extends HitsPerPageConnectorParams = HitsPerPageConnectorParams
-> = (
-  /**
-   * Render function for the custom **HitsPerPage** widget.
-   */
-  render: HitsPerPageRenderer<THitsPerPageWidgetParams>,
-
-  /**
-   * Unmount function called when the widget is disposed.
-   */
-  unmount?: Unmounter
-) => HitsPerPageWidgetFactory<THitsPerPageWidgetParams>;
-
-const connectHitsPerPage: HitsPerPageConnector = function connectHitsPerPage(
-  renderFn,
-  unmountFn = noop
-) {
+} & RendererOptions<THitsPerPageWidgetParams & HitsPerPageConnectorParams>;
+
+export type HitsPerPageRenderer<THitsPerPageWidgetParams> = Renderer<
+  HitsPerPageRendererOptions<
+    THitsPerPageWidgetParams & HitsPerPageConnectorParams
+  >
+>;
+
+export type HitsPerPageWidgetFactory<THitsPerPageWidgetParams> = WidgetFactory<
+  HitsPerPageConnectorParams & THitsPerPageWidgetParams
+>;
+
+export default function connectHitsPerPage<THitsPerPageWidgetParams = {}>(
+  renderFn: HitsPerPageRenderer<
+    HitsPerPageConnectorParams & THitsPerPageWidgetParams
```

Comment:
This is not correct. The `renderFn` doesn't receive `THitsPerPageWidgetParams`, only `HitsPerPageConnectorParams`. We can probably pass `THitsPerPageWidgetParams` in the widget file when calling `makeHitsPerPage<HitsPerPageWidgetOptions>({ items, transformItems })`.